<launch>
  <!-- HoloOcean Configuration -->
  <arg name="env_config" default="OceanSimple-Hovering2CameraOnly-VisibilityEstimation"/>
  <arg name="show_viewport" default="true"/>
  <arg name="fog_density" default="2.0"/>
  <arg name="render_quality" default="3"/>
  <arg name="trajectory_update_rate" default="10.0"/>
  <!-- Path to config files -->
  <arg name="pid_config" default="$(find holoocean_ros)/config/pid_config.yaml"/>
  <arg name="camera_intrinsics_file" default="$(find holoocean_ros)/config/depth_intrinsics.yaml"/>
  <arg name="left_camera_intrinsics_file" default="$(find holoocean_ros)/config/left_holoocean.yaml"/>
  <arg name="right_camera_intrinsics_file" default="$(find holoocean_ros)/config/right_holoocean.yaml"/>
  <!-- Launch the HoloOcean ROS node -->
  <node name="holoocean_integrated" pkg="holoocean_ros" type="holoocean_node_raft.py" output="screen">
    <param name="env_config" value="$(arg env_config)"/>
    <param name="show_viewport" value="$(arg show_viewport)"/>
    <param name="fog_density" value="$(arg fog_density)"/>
    <param name="render_quality" value="$(arg render_quality)"/>
    <param name="trajectory_update_rate" value="$(arg trajectory_update_rate)"/>
    <param name="pid_config" value="$(arg pid_config)"/>
    <param name="camera_intrinsics_file" value="$(arg camera_intrinsics_file)"/>
    <param name="left_camera_intrinsics_file" value="$(arg left_camera_intrinsics_file)"/>
    <param name="right_camera_intrinsics_file" value="$(arg right_camera_intrinsics_file)"/>
  </node>

    <!-- Start nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" output="screen"/>
  <!-- Start depth_image_proc nodelet -->
  <node pkg="nodelet" type="nodelet" name="depth_to_pointcloud"
        args="load depth_image_proc/point_cloud_xyz nodelet_manager" output="screen">
    <remap from="image_rect" to="/camera/depth/image_raw"/>
    <remap from="camera_info" to="/camera/depth/camera_info"/>
    <remap from="points" to="/camera/depth/points"/>
    <param name="sensor_frame" value="camera_depth_optical_frame" />
  </node>
</launch>
